teuthology/exceptions.py

```diff
@@ -51,7 +51,12 @@ class CommandFailedError(Exception):
     Exception thrown on command failure
     """
     def __init__(self, command, exitstatus, node=None, label=None):
-        self.command = command
+        if isinstance(command, (list, tuple)):
+            self.command = ' '.join(command)
```

Comment:
I think the right place for this transformation would be in the `__str__` method. It would improve readability, yes.

Reply:
True, that'll be much better.